### DISCUSSION TITLE
What are the LicenseStatus enum values?

### DISCUSSION THREAD

**Discussion by dhamelin** [KirillKuzyk/Google-Play-License-Check#12] [Q&A] [answered]
Hi

In code sample I see LICENSED / NOT_LICENSE, but is there a "NOT_YET_CHECKED". For use cases where the player is doing its first boot offline, I don't want to consider him non-legit before a first online check has been made.

I see that there are multiple other status (response code) possible: https://developer.android.com/google/play/licensing/licensing-reference

Thank you.

**Comment by KyryloKuzyk** [comment, marked as answer]
Hi,

> In code sample I see LICENSED / NOT_LICENSE, but is there a "NOT_YET_CHECKED". For use cases where the player is doing its first boot offline, I don't want to consider him non-legit before a first online check has been made.

The plugin provides the GooglePlayLicense.cachedLicense property. If this property is null, the license check has never been done before. After successfully connecting with the licensing server, the plugin will cache the license response, and the property will conveniently hold the last license check result.

Please note that the cached license is stored in PlayerPrefs, which is not secure. You can provide your own secure storage implementation with the help of the GooglePlayLicense.SetPersistentStorageImplementation() API.

> I see that there are multiple other status (response code) possible: https://developer.android.com/google/play/licensing/licensing-reference

All of the response codes are supported by the plugin. The plugin follows the official API as closely as possible.

Here is what the LicenseStatus enum looks like in the plugin. It contains all the response codes from the native implementation, plus the additional ERROR_INTERNAL code to represent the plugin's client-side errors (like an incorrect public key, etc.).
```
public enum LicenseStatus {
    LICENSED = 0x0,
    NOT_LICENSED = 0x1,
    LICENSED_OLD_KEY = 0x2,
    ERROR_NOT_MARKET_MANAGED = 0x3,
    ERROR_SERVER_FAILURE = 0x4,
    ERROR_OVER_QUOTA = 0x5,
    ERROR_CONTACTING_SERVER = 0x101, //257
    ERROR_INVALID_PACKAGE_NAME = 0x102, //258
    ERROR_NON_MATCHING_UID = 0x103, //259
    /// Check <see cref="License.debugInfo"/> for more details.
    ERROR_INTERNAL = -1
}
```
